### ISSUE TITLE
the time of dereplicate

### ISSUE THREAD

**Issue by MonaLiu421** [MrOlm/drep#133]
hi,when i use drep to deal with more than 3000+ bacteria geonomes，its too slow，i want to know if you have evaluated the upper limit of the number of genomes that the software worked

**Comment by MrOlm** [comment]
Hello,

I've used dRep with ~400,000 genomes before, but to do that you need to use greedy algorithms (https://drep.readthedocs.io/en/latest/choosing_parameters.html#using-greedy-algorithms). For 3000 genomes you shouldn't need to use greedy algorithms; I would suggest using fastANI if you're not already and giving the job more cores if possible.

Best,
MO

**Comment by MonaLiu421** [comment]
thanks for your suggestions, I will try it. And I wanna know the reason for this problem. I didn't use the ANImf ，and just the Mash algorithm(--SkipSecondary), Why it still doesn't work?

**Comment by MrOlm** [comment]
The distance matrix created by an all vs all of mash ends up being really really big, and that’s the problem. That’s why the “multiround primary clustering” argument fixes the problem

best,
Matt

**Comment by MonaLiu421** [comment]
ok, thanks! I get it, and I have tested the de-redundancy of more than 6000 genomes, used the  -p 16 -comp 50 -con 5 -nc 0.25 -pa 0.99 -sa 0.99 --S_algorithm fastANI --multiround_primary_clustering --greedy_secondary_clustering.but the following error is reported in the log file. But there is a result output. I don’t know if it is normal.
![image](https://user-images.githubusercontent.com/45484925/139802001-ab560d8b-5898-47cc-81bd-53aac1ce7da9.png)

**Comment by MrOlm** [comment]
Yeah that's normal- the plots break all the time due to changes in python dependencies over the years. It won't impact anything except the particular plot that the error is coming from

-MO

**Comment by MonaLiu421** [comment]
Ok, thanks. But another test reported an error, when I tested the de-redundancy of more than 6000 genomes(same data as above), I used the -p 16 -comp 50 -con 5 -nc 0.25 -pa 0.99  --multiround_primary_clustering --SkipSecondary  -d. The task error interrupt.
![image](https://user-images.githubusercontent.com/45484925/139973160-32e191f1-6674-4b90-80f6-2a240c68b904.png)
![image](https://user-images.githubusercontent.com/45484925/139973632-10a4e84b-91b3-4daa-907f-c2a8b3c29e2d.png)

**Comment by MrOlm** [comment]
Interesting. It's possible that there is bug that prevents `--SkipSecondary` and `--multiround_primary_clustering` from being run simultaneously. 

Could you please confirm that you're running dRep version 3.2.2? If so, could you please upload the log file (named something like `log/logger.log`)

**Comment by MrOlm** [comment]
If this is a bug, I also believe a workaround would be to add `-centW 0`

**Comment by MonaLiu421** [comment]
ok  i use version 3.0 now.you mean  the F*(centrality - S_ani) set 0？ can this sovled in version3.0？I think the software installation is a bit cumbersome, if this can be solved, that would be great

**Comment by MrOlm** [comment]
`pip install drep --upgrade` will install it. Alternatively `conda install -c bioconda drep`. 

If version 3.2.2 works, great and no need to change centW. If not, try setting it to 0

**Comment by MonaLiu421** [comment]
ok，thanks a million，you’re so nice

**Comment by MonaLiu421** [comment]
sorry to bother you again,it seems ok after using version 3.2.2. There is this sentence in the log file-"You skipped secondary clustering but have centrality weight above 0. You cant do that. I will now set the centrality weight to 0 to avoid a crash" .it‘s same as setting -centW 0.
and the following error about plotting is reported in the log file. 
![image](https://user-images.githubusercontent.com/45484925/140021480-0b7084ea-95ad-4f32-aad3-307b5e0a5e7e.png)
![image](https://user-images.githubusercontent.com/45484925/140021495-119fed6f-3577-4574-abce-755ac8a9b614.png)
I guess its normal? the same as above. and I don't known what it means？
![image](https://user-images.githubusercontent.com/45484925/140021831-9769c91e-5bfe-4f04-a4f2-74c02915e948.png)

**Comment by MrOlm** [comment]
Yep- those errrors are fine and should only impact the plots

**Comment by MonaLiu421** [comment]
ok，thanks a million，you’re so nice😊

**Comment by MonaLiu421** [comment]
hi,MrOlm, I‘m back.Recently, a problem has bothered me again. I want to know why I had to go to this point in the task, there was no movement.
![image](https://user-images.githubusercontent.com/45484925/143515550-7a8ee229-e313-4403-aa24-8db80d9ac940.png)
the Widb.csv file hasn’t  generated.
Weirdly,the whole task was not completed and the log file was incomplete. The task stopped. Although there are indeed non-redundant genomes selected, Will this affect my final result of de-replication？

**Comment by MrOlm** [comment]
Hello,

In this case the de-replicated genomes are fine- it’s just a problem generating the warnings.

Best,
Matt

**Comment by MonaLiu421** [comment]
But at this point, the task process has stopped, and the subsequent analysis and drawing steps have not been run. I don’t know why? the figure and widb.csv haven’t generated.
